website/server/models/group.js

```diff
@@ -577,6 +577,25 @@ schema.statics.cleanGroupQuest = function cleanGroupQuest () {
   };
 };
 
+async function _updateUserWithRetries (userId, updates) {
+  let data = {
+    saved: false,
+    retries: 0,
+  };
+  while (!data.saved && data.retries < 5) {
+    await User.update({_id: userId}, updates).exec((err, raw) => { // eslint-disable-line babel/no-await-in-loop
```

Comment:
I agree this was kind of messy and made the scopes hard to follow. I've updated this to a recursive method that I think is much cleaner.